ethcore/src/types/trace_types/filter.rs

```diff
@@ -108,27 +108,27 @@ impl Filter {
 
 	/// Returns true if given trace matches the filter.
 	pub fn matches(&self, trace: &FlatTrace) -> bool {
-		let action = match trace.action {
+		match trace.action {
 			Action::Call(ref call) => {
 				let from_matches = self.from_address.matches(&call.from);
 				let to_matches = self.to_address.matches(&call.to);
 				from_matches && to_matches
 			}
 			Action::Create(ref create) => {
 				let from_matches = self.from_address.matches(&create.from);
-				let to_matches = self.to_address.matches_all();
+
+				let to_matches = match trace.result {
```

Comment:
No. To be honest, there should be just one enum. Having two is just a performance workaround to avoid unnecessary copying data when trace is created.